backend/src/v2/metadata/client.go

```diff
@@ -978,6 +992,40 @@ func (c *Client) matchedArtifactOrNot(ctx context.Context, target *pb.Artifact,
 	return false, nil
 }
 
+// GetContexts ...
+func (c *Client) GetContexts(ctx context.Context, maxResultSize int32, orderByAscending bool, orderByField, filterQuery, nextPageToken string) ([]*pb.Context, *string, error) {
+	opts := buildListOpts(maxResultSize, orderByAscending, orderByField, filterQuery, nextPageToken)
+	req := &pb.GetContextsRequest{Options: opts}
+	res, err := c.svc.GetContexts(ctx, req)
+	if err != nil {
+		return nil, nil, err
+	}
+	return res.Contexts, res.NextPageToken, nil
+}
+
+// GetContextByArtifactID fetches the system.PipelineRun context for this Artifact ID
+func (c *Client) GetContextByArtifactID(ctx context.Context, id int64) (*pb.Context, error) {
```

Comment:
any concern about needing to get the Pipeline context for an artifact some day? If so, consider renaming this to `GetPipelineRunContextByArtifactID`